### PR TITLE
Credentials 6: persist credentials and add automation tests

### PR DESCRIPTION
**Pull request by ianwremmel**

### CODE REVIEW

File: packages/test-helper-automation/src/wd/print-logs.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    .log('browser')
+    .then(function doPrintLogs(logs) {
+      logs.forEach(function formatLog(log) {
+        try {
+          log.message = JSON.parse(log.message);
+          const method = console[log.message.message.level] || console.log;
+          method.call(console, 'browser log:', log.message.message.text);
+        }
+        catch (err) {
+          console.log('browser log:', log.message);
```
Comment (Josh-Dykstra) [inline packages/test-helper-automation/src/wd/print-logs.js]: Should we print the error here? I guess this is just falling back if the attempt above doesn't work. Maybe we don't care why it failed - would just pollute the logs to print it out.

File: packages/test-helper-automation/src/wd/print-logs.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+wd.addPromiseChainMethod('printLogs', function printLogs() {
+  return this
+    .log('browser')
+    .then(function doPrintLogs(logs) {
+      logs.forEach(function formatLog(log) {
+        try {
+          log.message = JSON.parse(log.message);
+          const method = console[log.message.message.level] || console.log;
+          method.call(console, 'browser log:', log.message.message.text);
```
Comment (Josh-Dykstra) [inline packages/test-helper-automation/src/wd/print-logs.js]: Is it really `log.message.message.text`?

File: packages/plugin-credentials/test/automation/spec/authorization-code-grant.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        .waitForElementByClassName(`ready`)
+        .title()
+          .should.eventually.become(`Authorization Automation Test`)
+        .waitForElementByCssSelector(`[title="Login with Authorization Code Grant"]`)
+          .click()
+        .login(user)
+        .waitForElementByClassName(`authorization-automation-test`)
+        .waitForElementById(`refresh-token`)
+          .text()
+            .should.eventually.not.be.empty
```
Comment (Josh-Dykstra) [inline packages/plugin-credentials/test/automation/spec/authorization-code-grant.js]: I guess we could make these checks a little more robust and compare them to the user's refresh token, but that's probably not necessary.

File: packages/plugin-credentials/test/automation/spec/authorization-code-grant.js [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+ * @private
+ */
+
+import {createBrowser} from '@ciscospark/test-helper-automation';
+import testUsers from '@ciscospark/test-helper-test-users';
+import pkg from '../../../package.json';
+
+const redirectUri = process.env.CISCOSPARK_REDIRECT_URI || process.env.REDIRECT_URI;
+
+describe(`spark-core`, function() {
```
Comment (Josh-Dykstra) [inline packages/plugin-credentials/test/automation/spec/authorization-code-grant.js]: Nice job on these tests.

File: packages/spark-core/src/plugins/credentials/credentials/browser.js [hunk 5]
```diff
@@ -12,17 +12,18 @@ import {assign, clone, has, omit, pick} from 'lodash';
...
+import SparkPlugin from '../../../lib/spark-plugin';
 
 /**
  * @private
  * @returns {undefined}
  */
 function noop() {/* eslint no-empty:[0] */}
 
-const Credentials = CredentialsBase.extend({
+const Credentials = SparkPlugin.extend(Object.assign(common, {
```
Comment (Josh-Dykstra) [inline packages/spark-core/src/plugins/credentials/credentials/browser.js]: Was this to make it play more nicely with the decorators using `@`, like `persist`?

Comment (Josh-Dykstra) [inline packages/spark-core/src/plugins/credentials/credentials/browser.js]: Ok. Now I see your commit message had another line 👍

**Comment by Josh-Dykstra** [comment]
Couple minor typos in docs and comments and some general questions. LGTM

File: packages/test-helper-automation/src/wd/print-logs.js (hunk 1 above)
Comment (ianwremmel) [inline packages/test-helper-automation/src/wd/print-logs.js]: it should only fail when JSON.parse fails, so the error isn't that important.

File: packages/plugin-credentials/test/automation/spec/authorization-code-grant.js (hunk 3 above)
Comment (ianwremmel) [inline packages/plugin-credentials/test/automation/spec/authorization-code-grant.js]: Yea, I'm actually thinking we'll need to `credentials-spec` package in the near future. this plugin does a lot of stuff that the credentials plugin in http-core doesn't need to, but both really need to have the same interface.

File: packages/test-helper-automation/src/wd/print-logs.js (hunk 2 above)
Comment (ianwremmel) [inline packages/test-helper-automation/src/wd/print-logs.js]: yep

**Pull request merged**